### PR TITLE
Refactor to use API in visual and graph modes

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Refactor the Visual and ASCII graph modes to use the API instead of `rz_core_cmd*()` calls.

**Test plan**

All green, try writing feature in both visual and panels modes.
Also try to run `afcf` command since the logic behind it was altered a bit.

**Closing issues**

Partially addresses https://github.com/rizinorg/rizin/issues/382

### CODE REVIEW

File: librz/core/cmd_seek.c [hunk 1]
```diff
@@ -171,8 +171,8 @@ static void cmd_seek_opcode(RzCore *core, const char *input, bool silent) {
 		n = 1;
 	}
 	int val = (n < 0)
-		? cmd_seek_opcode_backward(core, -n, silent)
-		: cmd_seek_opcode_forward(core, n, silent);
+		? rz_core_seek_opcode_backward(core, -n, silent)
```
Comment (ret2libc) [inline librz/core/cmd_seek.c]: I suggest to extract a `rz_core_seek_opcode` that calls the right function based on whether n > 0 or < 0.

File: librz/core/agraph.c [hunk 2]
```diff
@@ -3619,7 +3619,11 @@ static void graphNodeMove(RzAGraph *g, int dir, int speed) {
 	if (is_mini(g)) {
 		discroll += (delta * speed);
 	} else if (g->is_dis) {
-		rz_core_cmdf(core, "so %d", (delta * 4) * speed);
+		if (delta < 0) {
```
Comment (ret2libc) [inline librz/core/agraph.c]: Use `rz_core_seek_opcode` here.

File: librz/core/agraph.c [hunk 3]
```diff
@@ -4632,7 +4636,7 @@ RZ_API int rz_core_visual_graph(RzCore *core, RzAGraph *g, RzAnalysisFunction *_
 			break;
 		case 'j':
 			if (g->is_dis) {
-				rz_core_cmd0(core, "so 1");
+				rz_core_seek_opcode_forward(core, 1, false);
```
Comment (ret2libc) [inline librz/core/agraph.c]: Use `rz_core_seek_opcode`.

File: librz/core/agraph.c [hunk 4]
```diff
@@ -4645,7 +4649,7 @@ RZ_API int rz_core_visual_graph(RzCore *core, RzAGraph *g, RzAnalysisFunction *_
 			break;
 		case 'k':
 			if (g->is_dis) {
-				rz_core_cmd0(core, "so -1");
+				rz_core_seek_opcode_backward(core, 1, false);
```
Comment (ret2libc) [inline librz/core/agraph.c]: Use `rz_core_seek_opcode`

File: librz/core/visual.c [hunk 5]
```diff
@@ -2003,11 +2003,13 @@ static bool insert_mode_enabled(RzCore *core) {
 	case 'e':
 	case 'f':
 		if (__nib != -1) {
-			rz_core_cmdf(core, "wx %c%c @ $$+%d", __nib, ch, core->print->cur);
+			char hexpair[3] = { __nib, ch, 0 };
+			rz_core_write_hexpair(core, core->offset + core->print->cur, hexpair);
 			core->print->cur++;
 			__nib = -1;
 		} else {
-			rz_core_cmdf(core, "wx %c. @ $$+%d", ch, core->print->cur);
```
Comment (ret2libc) [inline librz/core/visual.c:2010]: Is `wx 9.` even a thing???

**Comment by Maijin** [comment]
Random thought - What about moving those ".", "]> #" and "(cursor)" as static const char or (in that situation) #define on top, like in disasm.c - that way if later we want to change the appearance or make it customizable it'll be easier for the refactoring?

File: librz/core/visual.c (hunk 5 above)
Comment (XVilka) [reply librz/core/visual.c:2010]: Nope, it returns the error.

File: librz/core/cmd_seek.c (hunk 1 above)
Comment (XVilka) [reply librz/core/cmd_seek.c]: Done.

File: librz/core/visual.c (hunk 5 above)
Comment (XVilka) [reply librz/core/visual.c:2010]: Should I remove this line?

Comment (ret2libc) [reply librz/core/visual.c:2010]: Yes, I think this line can be removed. Keep the `__nib = ch` part, that's necessary.
In visual mode you can write at minimum 1 byte.

**Review by wargio — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]
Awesome!

**Pull request merged**